Evaluation creation Updated

AltSumpreme opened the pull request:
## Changes Made

### - Changed PUT Request to POST
### - Added Foreign Key Constraint Error Handling
### - Made `projectId` in the Evaluation Model Unique
### - Added a composite key having both userId and projectId
### -User access revoked for GET/{id} route .
### - Deletion only deletes the evaluation done by the evaluator

AltSumpreme committed:
fix:changed the put request to post and projectId made unique in the schema

AltSumpreme committed:
fix:changed the prisma query to upsert

AltSumpreme committed:
feat:added the foreign key constraint

ThEditor commented on src/routes/evalutation/index.ts:
```diff
@@ -124,10 +123,10 @@ evaluationRouter.openapi(deleteEvaluation, async (ctx) => {
       return ctx.text("You do not have permission to delete evaluations", 403);
     }
-    const id = ctx.req.param("id");
+    const projectId = ctx.req.param("id");
```
`deleteEvaluation` should be deleting an evaluation and not all evaluations of a project.
Each project can have multiple evaluation, use `id` of Evaluation.

ThEditor commented on src/routes/evalutation/index.ts:
```diff
@@ -12,40 +12,39 @@ import { getCurrentUser } from "../../lib/auth-provider.js";
-    }
-
-    return ctx.json({ message: "Record Updated", evaluation }, 201);
+    const evaluation = await prisma.evaluation.upsert({
```
don't upsert, split into two endpoints - `createEvaluation` (insert & POST) and `updateEvaluation` (update & POST)

AltSumpreme committed:
feat(schema): add composite key (userId, projectId) and update access control

- Added a composite key consisting of `userId` and `projectId` in the schema.
- Removed user access for the `getById` route to enhance security and streamline access control.

AltSumpreme committed:
feat(schema):userschema in the schema folder updated
